Source/WebCore/style/StyleBuilderConverter.h

```diff
@@ -554,13 +554,23 @@ inline StyleSelfAlignmentData BuilderConverter::convertSelfOrDefaultAlignmentDat
 
     // Flip the position according to position-try fallback, if specified.
     if (auto positionTryFallback = builderState.positionTryFallback()) {
+        auto writingMode = builderState.style().writingMode();
         for (auto tactic : positionTryFallback->tactics) {
             switch (tactic) {
+            case PositionTryFallback::Tactic::FlipX:
+                if ((builderState.cssPropertyID() == CSSPropertyAlignSelf && writingMode.isHorizontal())
+                    || (builderState.cssPropertyID() == CSSPropertyJustifySelf && !writingMode.isHorizontal()))
```

Comment:
I think you can just write 
```suggestion
                if (writingMode.isHorizontal() ? (builderState.cssPropertyID() == CSSPropertyJustifySelf) : (builderState.cssPropertyID() == CSSPropertyJustifySelf))
```

Also I think your logic is backwards, horizontal flip-x should be flipping justifySelf, right?